### ISSUE TITLE
Thermal Evaporation Chamber is Extremely Laggy

### ISSUE THREAD

**Issue by benwornes** [mekanism/Mekanism#7201]
*Please use the search functionality before reporting an issue. Also take a look at the closed issues!*

#### Issue description:
I took a look at previously closed issues and it seems to have been an issue before. I have been playing the Seaopolis mod pack, and as soon as I fly up to the top of my two thermal evaporation chambers, I go from around 100 Fps to around 3 Fps. From what I have gathered, it seems to be an issue with spawnable blocks. As soon as I place a torch on top of the chambers, I don't get any lag at all.

#### Steps to reproduce:
1. Have two 18 tall Thermal Evaporation Chamber with 4 advanced solar generators each, two blocks apart.
2. Don't place torches on top - observe lag
3. Place torches on top - don't observe lag

#### Version (make sure you are on the latest version before reporting):
**Forge:** version 31.1.16
**Mekanism:** 1.16.5 version 10.0.21.448
**Other relevant version:** Seaopolis: version B5.1

#### If a (crash)log is relevant for this issue, link it here: (_It's almost always relevant_)

[[gist](https://gist.github.com/) / [pastebin](https://pastebin.com/) / etc link here. Please make sure that it isn't set to expire.]

**Comment by EterniaLogic** [comment]
Indeed, I have several thermal evaporation chambers right next to eachother and I'm getting 2 fps. when I'm at the solar-panel level. (within 10 blocks near solar panels)

![Imgur](https://i.imgur.com/pime2WY.png)

**Comment by cyberdragon442** [comment]
This also lags the server in multiplayer. The server I'm playing on just outlawed towers and made people use creative ender tanks because it got so bad.

**Comment by pupnewfster** [comment]
I mean I think it may be more the solar panels than the evap towers themselves, in which case just using heaters and not having the solar panels and having evaporation blocks there instead would make the lag not happen. Which imo is the better way to use towers anyways, though this is still on my list of things to look into.

**Comment by SotomaniaGr** [comment]
I have the exact same issue on Enigmatica 6 which has Mekanism 10.1.1, as a temporary fix, I placed torches on top as the op mentioned.

**Comment by pupnewfster** [comment]
I would be somewhat interested in a spark profile log (one both of the server and the client) without the torches to see if it shows what exactly is the part causing the lag.

**Comment by SotomaniaGr** [comment]
I did three spark profiles just in case, My results should be taken with a grain of salt as I am not sure whether I captured the lag correctly as it wasn't as constant, and I couldn't replicate it to the same degree I experienced it previously. Sorry for taking two hours, modpack took too long to open.
https://spark.lucko.me/xDPV1iYdxs
https://spark.lucko.me/2mPWZNsT7j
https://spark.lucko.me/fCROupSJoL

**Comment by mpavkovic** [comment]
For those having the same issue and coming to this ticket, the problem is caused by a mod "More Overlays Updated", just turn off the overlay (F7 by default) and the lag will disappear (or place a light source on top of Thermal Evaporation Plant as others suggested). My guess is that it is trying to draw an overlay inside the thermal evaporation plant and it kills the FPS - that's why it lags only when you're near the top of the plant and only if light level is below 7. I've opened a ticket on their github: https://github.com/r8420/MoreOverlays-updated/issues/87
